File: lista 15questoes/15 questoes em python/att11.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas_trabalhadas, valor_hora):
        self.horas_trabalhadas = horas_trabalhadas
        self.valor_hora = valor_hora

    def calcular_salario(self):
        return self.horas_trabalhadas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario):
        self.salario = salario

    def calcular_salario(self):
        return self.salario

funcionario1 = FuncionarioHorista(40, 50)
funcionario2 = FuncionarioAssalariado(3000)
print(funcionario1.calcular_salario())
print(funcionario2.calcular_salario())

File: lista 15questoes/15 questoes em java/att11.java
abstract class Funcionario {
    abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private int horasTrabalhadas;
    private double valorHora;

    public FuncionarioHorista(int horasTrabalhadas, double valorHora) {
        this.horasTrabalhadas = horasTrabalhadas;
        this.valorHora = valorHora;
    }

    double calcularSalario() {
        return horasTrabalhadas * valorHora;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salario;

    public FuncionarioAssalariado(double salario) {
        this.salario = salario;
    }

    double calcularSalario() {
        return salario;
    }
}

Funcionario funcionario1 = new FuncionarioHorista(40, 50);
Funcionario funcionario2 = new FuncionarioAssalariado(3000);
System.out.println(funcionario1.calcularSalario());
System.out.println(funcionario2.calcularSalario());
